modules/indexing/src/test/java/org/apache/ignite/internal/processors/cache/IgniteDynamicSqlRestoreTest.java

```diff
@@ -273,6 +275,103 @@ public void testIndexCreationWhenNodeStopped() throws Exception {
         }
     }
 
+    /**
+     * Tests a scenario when an index gets created when a node is turned off and no checkpoint is issued upon the node
+     * stop.
+     */
+    @WithSystemProperty(key = IGNITE_PDS_SKIP_CHECKPOINT_ON_NODE_STOP, value = "true")
+    @Test
+    public void testIndexCreationWhenNodeIsStoppedWithoutCheckpoint() throws Exception {
+        // Start topology.
+        Ignite ignite = startGrids(2);
+
+        ignite.cluster().active(true);
+
+        // Create table, add some data.
+        int entryCnt = 50;
+
+        try (Connection conn = DriverManager.getConnection("jdbc:ignite:thin://127.0.0.1:10800")) {
```

Comment:
What's the reason of using a JDBC here? Could this be replicated with "native" SQL API?